hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/ApiTraceGraph.java

```diff
@@ -38,25 +39,44 @@
   private final StructuredTrace trace;
   private final List<ApiNode<Event>> apiNodeList;
   private final List<ApiNodeEventEdge> apiNodeEventEdgeList;
+  // set of outbound edges for apiNode
+  private final Map<Integer, Set<Integer>> apiNodeIdxToEdges;
   private final Map<ByteBuffer, Integer> eventIdToIndexInTrace;
   // map of api node to index in api nodes list, since we want to build edges between api nodes
   private final Map<ApiNode<Event>, Integer> apiNodeToIndex;
   // map of entry boundary event to api node. each api node has one entry boundary event
   private final Map<Event, ApiNode<Event>> entryBoundaryToApiNode;
   private final Table<Integer, Integer, Edge> traceEdgeTable;
+  // set of exit boundary events of apiNode, with no outgoing edge to any apiNode
+  private final Set<Integer> apiExitBoundaryEventIdxWithNoOutgoingEdge;
+  // set of exit boundary events of apiNode, with outgoing edge to some apiNode
```

Comment:
Is it eventId (or an index into some other structure)? EventId should be bytes, right?

Reply:
It is index of eventId (in the eventList in StructuredTrace)